// fly-admin/dist/lib/secret.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Secret = void 0;
const setSecretsQuery = `mutation($input: SetSecretsInput!) {
  setSecrets(input: $input) {
    release {
      id
      version
      reason
      description
      user {
        id
        email
        name
      }
      evaluationId
      createdAt
    }
  }
}`;
const unsetSecretsQuery = `mutation($input: UnsetSecretsInput!) {
  unsetSecrets(input: $input) {
    release {
      id
      version
      reason
      description
      user {
        id
        email
        name
      }
      evaluationId
      createdAt
    }
  }
}`;
class Secret {
    constructor(client) {
        this.client = client;
    }
    setSecrets(input) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.client.gqlPostOrThrow({
                query: setSecretsQuery,
                variables: { input },
            });
        });
    }
    unsetSecrets(input) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.client.gqlPostOrThrow({
                query: unsetSecretsQuery,
                variables: { input },
            });
        });
    }
}
exports.Secret = Secret;
